### ISSUE TITLE
Solr docs generated by ActiveFedora and Hyrax::ValkyrieWorkIndexer are not compatible

### ISSUE THREAD

**Issue by elrayle** [samvera/hyrax#4487]
### Descriptive summary

I would expect the solr doc created by Hyrax::ValkyrieWorkIndexer to generate the same set of fields as the ActiveFedora generated solr doc.

### Rationale

With both processes producing the same solr doc fields, existing solr indices will not need to be migrated when switching from ActiveFedora to Valkyrie.

### Expected behavior

Generated fields would have the same field names and values.

### Actual behavior

![image](https://user-images.githubusercontent.com/6855473/91320709-d8ae6680-e78b-11ea-9ba5-a6d9e1061eef.png)

NOTE: Excluded all custom fields in our indexer.

### Steps to reproduce the behavior

In my local app, I created a custom indexer which inherits from Hyrax::ValkyrieWorkIndexer.  I use the following in a spec to create the publication work which is Valkyrie::Resource, save it, and index it.  

```
  let!(:publication) do
    pub = FactoryBot.valkyrie_create(:publicationr, identifier: identifier)
    solr_doc = PublicationrIndexer.new(resource: pub).to_solr
byebug
    Hyrax::SolrService.new(use_valkyrie: true).add(solr_doc, commit: true)
    pub
  end
```

* `valkyrie_create` saves the work through ActiveFedora creating the ActiveFedora solr doc.  
* `Hyrax::SolrService` saves the ValkyrieWorkIndexer generated solr doc to a separate solr core.

**Comment by elrayle** [comment]
Per discussion in Samvera-Tech call today, the diffs reflect work in progress.  The path forward is to create PRs that move toward making the Hyrax::ValkyrieWorkIndexer consistent with the AF generated solr doc.

**Comment by elrayle** [comment]
### Related Issues

PR #4780 Add indexing for Hyrax::FileSet through valkyrie indexing 

Issue #4795 - Hyrax::Works do not show Date Added in Dashboard -> Works
Issue #4788 - Hyrax::Works do not show thumbnail in Dashboard -> Works
Issue #4794 - Hyrax::Works do not index file sets

**Comment by elrayle** [comment]
This is mostly complete.  It appears that permissions is the only part that is not in the Valkyrie solr document.  I opened Issue #4901 to cover just that part.

New comparison of the solr docs created by ActiveFedora and Valkyrie indexing...
![image](https://user-images.githubusercontent.com/6855473/116748961-3a376c80-a9ce-11eb-9d7d-b9dca345222a.png)

Closing this issue.  See Issue #4901 for remaining issue with permissions.
